Controls mutex

Wigu701 opened the pull request:
Added mutex code

bennyE31 committed:
Initial Controls Structure created

VishKing17 committed:
Ported driver files from sc1-main-io

Commit:
Added ported files to QT config file

nkurszewski committed:
Changed filepaths in i2c.cpp and serial.cpp to allow Qt project to build properly. Fixed typo in solar-car-dashboard.pro.

bennyE31 committed:
Update i2c drivers

bennyE31 committed:
Ads template

bquan0 committed:
ads1219 loop function

bennyE31 committed:
Get this to compile :)

bquan0 committed:
changes implemented

nkurszewski committed:
Copied from sc1-main-io/embedded to sc1-driver-io/embedded. Adjusted include statements in copied files to prevent errors in Qt. Added filepaths for copied header and source files to solar-car-dashboard.pro.

bennyE31 committed:
fix some paths

bquan0 committed:
controls test loop

bquan0 committed:
test loop 2

bquan0 committed:
test loop 3

bquan0 committed:
test loop 4

bquan0 committed:
test loop 5

bquan0 committed:
test loop working

bquan0 committed:
ads1219 test

Wigu701 committed:
Untested implementation of controls writing to stream

Attempted implementation of writing to byte array with mutex.

Wigu701 committed:
Revision of mutex locking and unlocking

Moved mutex unlocking and unlocking to immediately before and after
writes the array to avoid time wasted holding mutex

Wigu701 committed:
controlsWrapper header file update and added sampling limiter

Added sleep to voltage reading loop so it samples at 10Hz

Added header guards to the controlsWrapper header file

Added dataReady() signal to controlsWrapper in event signal is needed to
indicate a read from the stream. Line to emit signal is currently
commented out

Wigu701 committed:
Added missing QMutex include

Wigu701 committed:
Cleaned up and debugging with controlsWrapper

Updated constructor calls to include update parameters

Attempted to revise constructor call and function calls for Ads1219
object. Compiler is no longer angry but no confidence in solution due to
inexperience with C

Added private method to header

Code now compiles

bennyE31 commented on embedded/controlsWrapper.h:
```diff
@@ -0,0 +1,28 @@
+        void addFloatToArray(QByteArray &dataArr, float data);
+
+        QByteArray &bytes;
+        QMutex mutex;
```
I think we also want the mutex to be passed in as a reference, similar to they byte array. Software will also need to access the same mutex, so it will need to be initialized outside of the class

Wigu701 committed:
Added mutex parameter to controlsWrapper constructor

Updated Constructor to include mutex reference to pass in

Updated the constructor call in DataUnpacker to reflect this

Note: Does not compile, DataUnpacker in this branch does not
declare/contain a mutex yet like it does in the main branch

Wigu701 committed:
Revisions to get controls mutex to work

Stopped DataGen from spitting into stream

Fixed improper pointer/reference/casting erors

Modified controlsWrapper to send dummy data
